node/pkg/governor/governor_prices.go

```diff
@@ -88,19 +88,30 @@ func createCoinGeckoQueries(idList []string, tokensPerQuery int) []string {
 	}
 
 	if ids != "" {
-		queries = append(queries, createCoinGeckoQuery(ids))
+		queries = append(queries, createCoinGeckoQuery(ids, coinGeckoApiKey))
 	}
 
 	return queries
 }
 
 // createCoinGeckoQuery creates a CoinGecko query for the specified set of IDs.
-func createCoinGeckoQuery(ids string) string {
+func createCoinGeckoQuery(ids string, coinGeckoApiKey string) string {
 	params := url.Values{}
 	params.Add("ids", ids)
 	params.Add("vs_currencies", "usd")
 
-	query := "https://api.coingecko.com/api/v3/simple/price?" + params.Encode()
+	// If modifying this code, ensure that the test 'TestCoinGeckoPriceChecks' passes when adding a pro API key to it.
+	// Since the code requires an API key (which we don't want to publish to git), this
+	// part of the test is normally skipped but mods to sensitive places should still be checked
+	query := ""
+	if coinGeckoApiKey == "" {
+		query = "https://api.coingecko.com/api/v3/simple/price?" + params.Encode()
+	} else { // Pro version API key path
+		query = "https://pro-api.coingecko.com/api/v3/simple/price?"
+		params.Add("x_cg_pro_api_key", coinGeckoApiKey)
+		query += "&" + params.Encode()
```

Comment:
You don't need the "&" here. The params encoding handles that for us.

I would add the param in the first line of the `else` block and then do the assignment to query once (like we did before and how it's done when no api key is provided)